src/axom/multimat/docs/sphinx/multimat_field_concepts.rst

```diff
@@ -0,0 +1,118 @@
+|                    | dense storage). This mapping allows materials within a   |
+|                    | cell to have their own unique values, which makes them   |
+|                    | useful for tracking data at a sub-cell level.            |
++--------------------+----------------------------------------------------------+
+
+#######################
+Data Layout
+#######################
+
+Simulation codes contain a variety of algorithms that may have a preference for how
+data are arranged in memory to ensure good performance. MultiMat supports
+fields with a ``PER_CELL_MAT`` mapping and there are two ways to organize such data.
+Fields are said to be **Cell-Dominant** (``CELL_DOM``) if they are stored such that
+each cell stores all of its material data to memory before proceeding to data for
+the next cell. Fields are **Material-Dominant** (``MAT_DOM``) if the data for all
+cells that use the material is stored before proceeding to the next material.
+The data layout for multi-material data can be thought of as 2 nested for-loops where
+the outer loop is the dominant loop. For example, if iterating over materials and
+then cells, the data are stored using ``MAT_DOM`` layout.
+
++--------------------+----------------------------------------------------------+
+| DataLayout         | Meaning                                                  |
++====================+==========================================================+
+| CELL_DOM           | Data are stored for each cell and then for each material |
+|                    | like this *(c=cell, m=material)*:                        |
+|                    |                                                          |
+|                    | ``{c0m0, c0m1, c0m2, ..., c1m0, c1m1, c1m2, ...}``       |
++--------------------+----------------------------------------------------------+
+| MAT_DOM            | Data are stored for each material and then for each cell |
+|                    | like this *(c=cell, m=material)*:                        |
+|                    |                                                          |
+|                    | ``{m0c0, m0c1, m0c2, m0c3, ... , m1c0, m1c1, m1c2, ...}``|
++--------------------+----------------------------------------------------------+
+
+#######################
+Sparsity Layout
+#######################
+
+Sparsity primarily concerns fields with ``PER_CELL_MAT`` mapping. When initializing
+the MultiMat object, the CMR indicates how materials are distributed
+over the mesh. It is completely acceptable for materials to skip over certain cells,
+which makes sense if we think about materials as a way to divide up the mesh into
+various regions or parts. There are ncells*nmats pairs of data that could be entered
+for MultiMat fields. For ``DENSE`` fields, the field must contain ncells*nmats values,
+with values present for cell/material pairs even where the material is not present.
```

Comment:
I'd recommend "with values present for cell/material pairs whether material is present or not."

Reply:
Changed.